Meta/Lagom/CMakeLists.txt

```diff
@@ -51,281 +79,459 @@ set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${LINKER_FLAGS}")
 set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} ${LINKER_FLAGS}")
 set(CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULE_LINKER_FLAGS} ${LINKER_FLAGS}")
 
+# FIXME: This is a hack, because the lagom stuff can be build individually or
+#        in combination with the system, we generate two Debug.h files. One in
+#        Build/AK/Debug.h and the other in Build/Meta/Lagom/AK/Debug.h.
+configure_file(../../AK/Debug.h.in AK/Debug.h @ONLY)
+configure_file(../../Kernel/Debug.h.in Kernel/Debug.h @ONLY)
+
 include_directories(../../)
 include_directories(../../Userland/)
 include_directories(../../Userland/Libraries/)
 include_directories(${CMAKE_BINARY_DIR})
 include_directories(${CMAKE_CURRENT_BINARY_DIR})
 
-if (BUILD_LAGOM AND NOT ENABLE_OSS_FUZZ AND NOT ENABLE_FUZZER_SANITIZER)
-    if (CMAKE_SOURCE_DIR MATCHES ".*/Lagom")
-        set(write_if_different ${CMAKE_SOURCE_DIR}/../write-only-on-difference.sh)
-        add_subdirectory(../../Userland/Libraries/LibUnicode/CodeGenerators ${CMAKE_CURRENT_BINARY_DIR}/LibUnicode/CodeGenerators)
-    endif()
+# install rules, think about moving to its own helper cmake file
+# Don't install Lagom libs into the target Root/
+# FIXME: Remove this check for 4594
+if (CMAKE_SOURCE_DIR MATCHES ".*/Lagom")
+    include(CMakePackageConfigHelpers)
+    include(GNUInstallDirs)
+
+    set(package Lagom)
```

Comment:
The comment that was above this line might be useful for the uninitiated in this project as well.

Reply:
👍  Good point, I've copied it over from one of your generated projects.